src/subsidy.cpp

```diff
@@ -383,15 +383,21 @@ bool FindSubsidyIndustryCargoRoute()
 	CargoID cid;
 
 	/* Randomize cargo type */
-	if (src_ind->produced_cargo[1] != CT_INVALID && HasBit(Random(), 0)) {
-		cid = src_ind->produced_cargo[1];
-		trans = src_ind->last_month_pct_transported[1];
-		total = src_ind->last_month_production[1];
-	} else {
-		cid = src_ind->produced_cargo[0];
-		trans = src_ind->last_month_pct_transported[0];
-		total = src_ind->last_month_production[0];
+	int num_cargos = 0;
+	for (size_t ci = 0; ci < lengthof(src_ind->produced_cargo); ci++) {
+		if (src_ind->produced_cargo[ci] != CT_INVALID) num_cargos++;
 	}
+	if (num_cargos == 0) return false; // industry produces nothing
+	int cargo_num = RandomRange(num_cargos) + 1;
+	int cargo_index;
+	for (cargo_index = 0; cargo_index < lengthof(src_ind->produced_cargo); cargo_index++) {
```

Comment:
This algorithm feels overly complicated. Does it seem right?

Reply:
"ci" and "cargo_index" for the same thing is a bit weird.
But otherwise I see nothing wrong.

Reply:
warning: comparison between signed and unsigned integer expressions [-Wsign-compare]